### ISSUE TITLE
GEOSERVER_DATA_DIR  mount on AKS, no rights

### ISSUE THREAD

**Issue by nicjanic** [kartoza/docker-geoserver#246]
Hello there,

It appears that if you use an AKS volume for the GEOSERVER_DATA_DIR, on workload startup, you don't have suffisant rights (chown occurs in the Docker build, but is overrided on container startup.

The solution could be to add an initContainer to chown the GEOSERVER_DATA_DIR using GEOSERVER_UID=1000 &
GEOSERVER_GID=10001

**Comment by NyakudyaA** [comment]
hi @nicjanic , Can you do a PR with your suggestion

**Comment by nicjanic** [comment]
For now, there is no Helm or K8S flavors on the project.

On our Workload, we have added this initContainer to ensure that the permission have been set correctly after the volume is mounted.

```
initContainers:
      - command:
        - sh
        - -c
        - chown -R 1000:1001 /opt/geoserver/data_dir
        image: busybox
        imagePullPolicy: Always
        name: volume-prewarming
        volumeMounts:
        - mountPath: /opt/geoserver/data_dir
          name: geoserver-conf2
      restartPolicy: Always
      schedulerName: default-scheduler
      securityContext: {}
      terminationGracePeriodSeconds: 30
      volumes:
      - name: geoserver-conf
        persistentVolumeClaim:
          claimName: geoserver-conf
```

**Comment by NyakudyaA** [comment]
@nicjanic Reference to the helm charts are here https://github.com/kartoza/docker-geoserver/tree/develop#kubernetes-helm-charts

I will look at this issue when I have a spare moment

**Comment by Aldiuser** [comment]
Hello guys,
the container is running with user something like geouser... This user has the id 1000. You must change the group in the securityContext of your deployment.yaml. This way you can skip creating an init container.

```yaml
securityContext:
  fsGroup: 1000
```

Now your directory could be opened by the user which is running in the container.

Full:

```yaml
  template:
    metadata:
      labels:
        mylabel: label
    spec:
      securityContext:
        fsGroup: 1000
      containers:
        - name: "geoserver"
```

**Comment by NyakudyaA** [comment]
This should be fixed in 2.21.1 chown occurs in entrypoint script
